src/test/kotlin/nl/info/zac/admin/ZaaktypeConfigurationServiceTest.kt

```diff
@@ -0,0 +1,148 @@
+import java.net.URI
+import java.util.UUID
+
+class ZaaktypeConfigurationServiceTest : BehaviorSpec({
+    val zaaktypeUri = URI("https://example.com/zaaktype/${UUID.randomUUID()}")
+
+    val entityManager = mockk<EntityManager>()
+    val ztcClientService = mockk<ZtcClientService>()
+    val zaaktypeCmmnConfigurationBeheerService = mockk<ZaaktypeCmmnConfigurationBeheerService>()
+    val zaaktypeBpmnConfigurationBeheerService = mockk<ZaaktypeBpmnConfigurationBeheerService>()
+    val zaaktypeConfigurationService = ZaaktypeConfigurationService(
+        entityManager,
+        ztcClientService,
+        zaaktypeCmmnConfigurationBeheerService,
+        zaaktypeBpmnConfigurationBeheerService
+    )
+
+    Given("a concept zaaktype event") {
+        val zaaktype = createZaakType(concept = true)
+
+        every { ztcClientService.clearZaaktypeCache() } returns ""
+        every { ztcClientService.readZaaktype(zaaktypeUri) } returns zaaktype
+
+        When("updating zaakafhandel parameters") {
+            zaaktypeConfigurationService.updateZaakafhandelParameters(zaaktypeUri)
+
+            Then("no update is actually made") {
+                verify(exactly = 0) {
+                    zaaktypeCmmnConfigurationBeheerService.upsertZaaktypeCmmnConfiguration(zaaktype)
+                    zaaktypeBpmnConfigurationBeheerService.copyConfiguration(zaaktype)
+                }
+            }
+        }
+    }
+
+    Given("a new version of zaaktype without existing zaakafhandelparameters") {
+        val zaaktype = createZaakType()
+
+        every { ztcClientService.clearZaaktypeCache() } returns ""
+        every { ztcClientService.readZaaktype(zaaktypeUri) } returns zaaktype
+
+        // Relaxed entity manager mocking; criteria queries and persisting
+        val criteriaQuery = mockk<CriteriaQuery<ZaaktypeConfiguration>>(relaxed = true)
+        every { entityManager.criteriaBuilder } returns mockk(relaxed = true) {
+            every { createQuery(ZaaktypeConfiguration::class.java) } returns criteriaQuery
+        }
+        every { entityManager.createQuery(criteriaQuery) } returns mockk {
+            every { setMaxResults(1) } returns this
+            every { resultList } returns emptyList()
+        }
+
+        When("updating zaakafhandel parameters") {
+            zaaktypeConfigurationService.updateZaakafhandelParameters(zaaktypeUri)
+
+            Then("no update is actually made") {
+                verify(exactly = 0) {
+                    zaaktypeCmmnConfigurationBeheerService.upsertZaaktypeCmmnConfiguration(zaaktype)
+                    zaaktypeBpmnConfigurationBeheerService.copyConfiguration(zaaktype)
+                }
+            }
+        }
+    }
+
+    Given("a new version of zaaktype with existing CMMN zaakafhandelparameters") {
+        val zaaktype = createZaakType()
+        val zaaktypeCmmnConfiguration = createZaaktypeCmmnConfiguration()
+
+        every { ztcClientService.clearZaaktypeCache() } returns ""
```

Comment:
Empty string as a result? Sounds like we could improve that function? Seems a bit weird no?